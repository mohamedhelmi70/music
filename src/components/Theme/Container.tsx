import React, {ReactNode} from 'react';
import {ActivityIndicator, StatusBar, StyleProp, StyleSheet, ViewStyle} from 'react-native';
import {View} from '../Themed';
import {colors} from '../../constants';

function Container({children, loading, style}: {children: ReactNode; loading?: boolean; style?: StyleProp<ViewStyle>}) {
    return (
        <View style={[styles.container, style]}>
            <StatusBar backgroundColor={colors.light.tint} barStyle="dark-content" />
            {loading ? <ActivityIndicator size="small" color={colors.light.primary} /> : children}
        </View>
    );
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        width: '100%',
        backgroundColor: colors.light.background,
    },
});

export default Container;
