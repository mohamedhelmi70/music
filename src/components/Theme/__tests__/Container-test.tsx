import React from 'react';
import renderer from 'react-test-renderer';
import {View} from '../../Themed';
import Container from '../Container';

// Describing a test suite
describe('<Container />', () => {
    test('Renders Container correctly', () => {
        const tree = renderer.create(<Container loading={false} children={<View />} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
