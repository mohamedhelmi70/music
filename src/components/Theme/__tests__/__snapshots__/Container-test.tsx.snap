// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Renders Container correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#F7F7F9",
      },
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#F7F7F9",
          "flex": 1,
          "justifyContent": "center",
          "width": "100%",
        },
        undefined,
      ],
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#F7F7F9",
        },
        undefined,
      ]
    }
  />
</View>
`;
